lighting-estimation-explainer.md

```diff
@@ -0,0 +1,91 @@
+# WebXR Device API - Lighting Estimation
+This document explains the portion of the WebXR APIs that enable developers to render augmented reality content that reacts to real world lighting.
+
+## Introduction
+
+"Lighting Estimation" is implemented by AR platforms using a combination of sensors, cameras, algorithms, and machine learning.  Lighting estimation provides input to rendering algorithms and shaders to ensure that the shading, shadows, and reflections of objects appear natural when presented in a diverse range of settings.
+
+The XRLightProbe and XRReflectionProbe interfaces expose the values that the platform offer to WebXR rendering engines.  Their corresponding accessor functions, XRFrame.getGlobalLightEstimate() and XRFrame.getGlobalReflectionProbe() are only accessible once the first frame of an AR session has started.  The promises may be resolved on the same frame or multiple frames later, depending on the platform capabilities.  In some cases, the promises may fail, indicating that the lighting values are not available at this time and should be requested again at a later time.
```

Comment:
Can the page tell the difference between failure that might succeed later or failures that indicate that this platform simply doesn't support this feature for say, reflections?

Reply:
Instead of the promises failing if the lighting has not been calculated yet, could we just specify a standard set of values to return instead? That would save every app from having to create their own fallback lighting, and it would aid with consistency.